math/krishnamurthy_test.go

```diff
@@ -0,0 +1,51 @@
+package math
+
+import (
+	"fmt"
+	"testing"
+)
+
+func retCases() []struct {
+	input        int64
+	output       bool
+	outputString string
+} {
+	return []struct {
+		input        int64
+		output       bool
+		outputString string
+	}{
+		{-3112312321, false, "is not"},
+		{0, false, "is not"},
+		{1, true, "is"},
+		{2, true, "is"},
+		{109, false, "is not"},
+		{145, true, "is"},
+		{943, false, "is not"},
+		{6327, false, "is not"},
+		{40585, true, "is"},
+		{9743821, false, "is not"},
+		{3421488712, false, "is not"},
+	}
+}
+
+func TestIsKrishnamurthyNumber(t *testing.T) {
+	for _, test := range retCases() {
+		t.Run(fmt.Sprintf("%d %s a Krishnamurthy Number", test.input, test.outputString), func(t *testing.T) {
+			res := IsKrishnamurthyNumber(test.input)
+			if res != test.output {
+				t.Errorf("For input %d ,Expected: %t ,Found: %t", test.input, test.output, res)
```

Comment:
This is minor and I'm approving it anyways:

The error strings in Go are usually kept lowercase since we can wrap these errors into other errors and it doesn't make sense to have uppercase words in the middle (unless we're talking about nouns).

Reply:
```suggestion
				t.Errorf("for input %d, expected: %t, found: %t", test.input, test.output, res)
```

Like so ^

Also, the comma's are in the wrong place.